servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -171,6 +172,17 @@ public Inode getInodeByPath(TachyonURI path) throws InvalidPathException {
   }
 
   /**
+   * @param path the path to get the inodes list for
+   * @return the inodes list with the given path
+   * @throws InvalidPathException if the path is invalid
+   */
+  public List<Inode> collectInodes(TachyonURI path) throws InvalidPathException {
```

Comment:
For example, a path `/a/b/c/d`, if
* `/c/d` does not exist, return a list composed of ["", a, b]
* all exist, return a list composed of ["", a, b, c, d]

For the 2nd case, not sure this name is suitable?